### PR TITLE
add screen_addr pointer txt.io

### PR DESCRIPTION
**Pull request by ClaudioDaffra**
Good morning

I have edited some files in the "txt" library.
I have provided the possibility of being able to change the position of the screen
working on different banks.

        ; ---- screen , color address
        %asm {{
            .weak
            screen_addr   =   $0400
            .endweak
            color_addr    =   $D800
        }}

I also adapted the following subroutines to change the screen address :

    asmsub  setcol  (ubyte col @X, ubyte row @Y, ubyte color @A) clobbers(A, Y)  {

    asmsub  getcol  (ubyte col @A, ubyte row @Y) clobbers(Y) -> ubyte @ A {

    sub  setcc  (ubyte column, ubyte row, ubyte char, ubyte charcolor)  {

I also added the possibility from the assembler to be able to use these addresses as pointers :

        %asm{{
        
            lda #$04
            inc txt.setchr._mod+1
            jsr txt.setchr._mod 

            lda #$05
            inc txt.setcol._mod+1
            jsr txt.setcol._mod 
            
        }}

below is a complete example of the operation, it is possible to change the position of the screen with the -D option already provided.
file esempio  :

    %import textio
    %zeropage basicsafe

    main {

        sub start() {

            txt.clear_screen() 

            txt.print("hello world !")
            
            txt.setchr(0,0,3)
            
            txt.setcol(0,0,2)
             
            %asm{{
            
                lda #$04
                inc txt.setchr._mod+1
                jsr txt.setchr._mod 

                lda #$05
                inc txt.setcol._mod+1
                jsr txt.setcol._mod 
                
            }}
             
            txt.setcc( 2,0,'.',2 );
            
            txt.nl()
        }
    }

Kind Regards
Claudio

### CODE REVIEW

File: compiler/res/prog8lib/c64/textio.p8 [hunk 1]
```diff
@@ -468,6 +468,14 @@ asmsub  input_chars  (uword buffer @ AY) clobbers(A) -> ubyte @ Y  {
 	}}
 }
 
+; ---- screen , color address
+%asm {{
+    .weak
+    screen_addr   =   $0400
+    .endweak
```
Comment (irmen) [inline compiler/res/prog8lib/c64/textio.p8:475]: should this .endweak not be after color_addr?

File: compiler/res/prog8lib/c64/textio.p8 [hunk 2]
```diff
@@ -484,10 +492,37 @@ asmsub  setchr  (ubyte col @X, ubyte row @Y, ubyte character @A) clobbers(A, Y)
...
+_mod		
+        sta  $ffff		; modified
+		rts
+
+_screenrows	.word  screen_addr + range(0, 1000, 40)
+_ptr    = setchr._mod +1
+	}}
+}
+
+asmsub  setcol  (ubyte col @X, ubyte row @Y, ubyte color @A) clobbers(A, Y)  {
```
Comment (irmen) [inline compiler/res/prog8lib/c64/textio.p8:504]: there's already setclr() that does the same thing, this new routine seem redundant?

File: compiler/res/prog8lib/c64/textio.p8 [hunk 3]
```diff
@@ -556,30 +611,9 @@ _mod		lda  $ffff		; modified
...
-		bcc  +
-		inc  _charmod+2
-		inc  _colormod+2
-+		lda  char
-_charmod	sta  $ffff		; modified
-		lda  charcolor
-_colormod	sta  $ffff		; modified
-		rts
-	}}
+    setchr( column , row, char )
```
Comment (irmen) [inline compiler/res/prog8lib/c64/textio.p8:614]: There's a problem with this refactoring, the routine now has to calculate the memory offsets twice, which slows down the routine. Is this acceptable...
I guess if we don't do this your other modifications don't work

**Review by irmen — CHANGES_REQUESTED** [review]

File: compiler/res/prog8lib/c64/textio.p8 (hunk 1 above)
Comment (ClaudioDaffra) [reply compiler/res/prog8lib/c64/textio.p8:475]: the color memory of the commodore 64 is in $ d800 cannot be changed

File: compiler/res/prog8lib/c64/textio.p8 (hunk 3 above)
Comment (ClaudioDaffra) [reply compiler/res/prog8lib/c64/textio.p8:614]: yes calculate twice, when use diffrerent banks (1-4) or differente memory location for screen character pointer.

Comment (ClaudioDaffra) [reply compiler/res/prog8lib/c64/textio.p8:614]: maybe better to keep the original setcc, for more speed and add 2 subroutines that calculate the offset of the screen if it is in different position.

File: compiler/res/prog8lib/c64/textio.p8 (hunk 2 above)
Comment (ClaudioDaffra) [reply compiler/res/prog8lib/c64/textio.p8:504]: yes redundant

**Comment by irmen** [comment]
I  appreciate the ideas and contributions

**Pull request closed without merging**